src/core/server/elasticsearch/version_check/ensure_es_version.ts

```diff
@@ -59,12 +61,13 @@ function getHumanizedNodeName(node: NodeInfo) {
 export function mapNodesVersionCompatibility(
   nodesInfo: NodesInfo,
   kibanaVersion: string,
-  ignoreVersionMismatch: boolean
+  ignoreVersionMismatch: boolean,
+  nodesInfoError?: UnauthorizedError | Error
 ): NodesVersionCompatibility {
-  if (Object.keys(nodesInfo.nodes ?? {}).length === 0) {
+  if (Object.keys(nodesInfo.nodes ?? {}).length === 0 && nodesInfoError) {
```

Comment:
If nodes are empty but we don't have `nodesInfoError`, it won't enter this if. For safety, I'd rather add 

```suggestion
  if (Object.keys(nodesInfo.nodes ?? {}).length === 0) {
```

```typescript
const extendedMsg = getExtendedErrorMessage(nodesInfoError);
...
message: `Unable to retrieve version information from Elasticsearch nodes${extendedMsg}`,

function getExtendedErrorMessage(error?: Error) {
  // handle all the ways to deconstruct the error (extract `.message`, `.reason` if available)...
}
```

What do you think?